cantools/subparsers/generate_c_source.py

```diff
@@ -1,4 +1,5 @@
 import os
+import os.path
```

Comment:
It should be just fine without this import.

Reply:
'Fine', yes.  Good, no.  Without importing the module you use you are depending on other code elsewhere having imported it for you.  This means that this file can break without it having made any changes.  In this case, admittedly, `os` itself does import `os.path` for you.  Still not a good habit.

Reply:
Not the best argument as it is a comment, but: https://github.com/python/cpython/blob/3.9/Lib/os.py#L3-L5

Reply:
Justifying exceptions to good habits is way more hassle than a consistent approach to imports.  You are concerned about the cost of updating a classifier when the unplanned Python 4 is released.  I expect it is much more effort to keep up with which packages promise to export their subpackages and modules not to mention tracking when they change those promises.